src/server/scripts/World/game_event_quest_cleanup.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2008-2019 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <ScriptMgr.h>
+#include <Player.h>
+#include <GameEventMgr.h>
+
+enum GameEvents : uint16
+{
+    GAME_EVENT_MIDSUMMER_FIRE_FESTIVAL    = 1,
+    GAME_EVENT_WINTER_VEIL                = 2,
+    GAME_EVENT_LUNAR_FESTIVAL             = 7,
+    GAME_EVENT_LOVE_IS_IN_THE_AIR         = 8,
+    GAME_EVENT_NOBLEGARDEN                = 9,
+    GAME_EVENT_CHILDRENS_WEEK             = 10,
+    GAME_EVENT_HARVEST_FESTIVAL           = 11,
+    GAME_EVENT_HALLOWS_END                = 12,
+    GAME_EVENT_BREWFEST                   = 24,
+    GAME_EVENT_PILGRIMS_BOUNTY            = 26,
+    GAME_EVENT_DAY_OF_THE_DEAD            = 51
+};
+
+class GameEventQuestCleanupScript : public PlayerScript
+{
+    public:
+        GameEventQuestCleanupScript() : PlayerScript("GameEventQuestCleanupScript") { }
+
+        void OnLogin(Player* player, bool /*firstLogin*/) override
```

Comment:
if we choose to keep the OnLogin() PlayerScript (not that each PlayerScript has a lot of hooks and even if empty, they add up), I would like you to profile a bit this code and find the most performant way to achieve the same result.
Maybe iterating the active player quests first, then check the game event mgr ? or maybe that's slow ? I have no clue.

Reply:
any follow up on this ?